### PR TITLE
feat : Implement Clipart (Drawables) dropdown area

### LINKED ISSUE #240: Implement Clipart (Drawables) dropdown area
The Cliparts are currently movable to the right and left. Instead there should be an area below and in future the user should be able to add their own cliparts.

Some inspiration from Whatsapp:
Screenshot from 2019-05-01 17-51-20

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #240

Changes:
- Clipart section is now a list with 6 columns

Screenshots for the change:
iss_240

### CODE REVIEW

**Comment by adityastic** [comment]
@yashk2000 @mariobehling @iamareebjamal Please Review

**Comment by mariobehling** [comment]
Please increase the size of the clipart whereever possible. At the moment some are difficult to see as they are a bit small.

**Comment by adityastic** [comment]
@mariobehling give me a minute

**Comment by adityastic** [comment]
@mariobehling updated the screenshot as well as changed the columns to 5

**Comment by mariobehling** [comment]
Please increase the size of the icons itself inside the white boxes, e.g. make the heart itself bigger.

**Comment by adityastic** [comment]
> Please increase the size of the icons itself inside the white boxes, e.g. make the heart itself bigger.

Sorry for misunderstanding, I got your point. Let me write the algorithm that will scale the images accordingly

**Comment by adityastic** [comment]
@mariobehling Please Look at the new Updated Screenshot Sir. The Algorithm is designed to remove the transparent pixels and scale them to fitCenter inside the White Box

File: app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt [hunk 1]
```diff
@@ -50,12 +53,70 @@ class DrawableAdapter(private val context: Context?, private val list: List<Draw
...
                 selectedPosition != -1 && selectedPosition == adapterPosition -> context?.resources?.getDrawable(R.color.colorAccent)
                 else -> context?.resources?.getDrawable(android.R.color.transparent)
             }
+            image.setColorFilter((when {
+                selectedPosition != -1 && selectedPosition == adapterPosition -> context?.resources?.getColor(android.R.color.white)
+                else -> context?.resources?.getColor(android.R.color.black)
+            }) ?: Color.parseColor("#000000"))
+        }
+
+        fun trim(source: Bitmap): Bitmap {
```
Comment (iamareebjamal) [inline app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: Should be no need of any algorithm, just reduce the padding on the button. If it is 0, apply negative margin

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: @iamareebjamal ok sir, let me try this as soon as I reach home

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: @iamareebjamal, Btw sir there is no padding given to the button. The drawables are in 40x11. The larger width was the issue which caused the image to fit inside the imageview with the transparent pixels. These transparent pixels made the image look smaller

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: @iamareebjamal I will still have to calculate the amount of margin to reduce for every bitmap as the transparent pixels vary from image to image

Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: @iamareebjamal sir, If you look in the current screenshot. Check the clipart at (2,1). It is much wider than the rest. All of them were the same with transparent pixels on all sides, which looked uneven. So I just trimmed those transparent pixels and then set the bitmap.

**Comment by mariobehling** [comment]
The resolution of some images is too small and the icons appear unsharp therefore. You probably need to recreate the images. Just scaling them to another resolution will not solve the problem.

**Comment by adityastic** [comment]
@mariobehling yes sir. The images are only 11 pixels tall. And that is the resolution supported by our badge

**Comment by adityastic** [comment]
@mariobehling I'm traveling right now, will reach by tomm morning. I will resize them to a large scale which will look appealing

**Comment by adityastic** [comment]
@mariobehling Good Morning Sir, I've updated the algorithm thanks for the suggestion they look beautiful now. Please Review :)

File: app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt (hunk 1 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: @iamareebjamal good morning sir, as the padding was 0, I tried applying negative margin of -20dp, It did zoom in but the images which were wider had its end chopped off. So if we plan to give negative margin, it needs to be dynamic according to the transparent pixels for which we need to create an algorithm anyways. After giving the negative margin, the resulting image will be too smooth and hence, we still need to perform a scale which I just added to this function. Please let me know your thoughts on this.
Regards

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Comment by adityastic** [comment]
@mariobehling just a min

File: app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt (hunk 1 above)
Comment (adityastic) [reply app/src/main/java/com/nilhcem/blenamebadge/adapter/DrawableAdapter.kt]: @iamareebjamal please review

**Comment by adityastic** [comment]
@mariobehling Please Review

**Comment by mariobehling** [comment]
Please follow up here and add more smiley, clock times, thumbs etc. https://github.com/fossasia/badge-magic-android/issues/268

**Comment by adityastic** [comment]
@mariobehling this is a fun one, I'd love to design more cliparts. Thanks for the opportunity

**Pull request merged**